python.d/monit.chart.py

```diff
@@ -0,0 +1,150 @@
+
+# charts order (can be overridden if you want less charts, or different order)
+ORDER = [ 'filesystem', 'directory', 'file', 'process', 'process_uptime', 'process_threads', 'process_children', 'host', 'host_latency', 'system', 'fifo', 'program', 'net' ]
+CHARTS = {
+    # id: {
+    #     'options': [name, title, units, family, context, charttype],
+    #     'lines': [
+    #         [unique_dimension_name, name, algorithm, multiplier, divisor]
+    #     ]}
+    'filesystem': {
+        'options': ['filesystems', 'Filesystems', 'filesystems', 'filesystem', 'monit.filesystems', 'line'],
+        'lines': []
+    },
+    'directory': {
+        'options': ['directories', 'Directories', 'directories', 'filesystem', 'monit.directories', 'line'],
+        'lines': []
+    },
+    'file': {
+        'options': ['files', 'Files', 'files', 'filesystem', 'monit.files', 'line'],
+        'lines': []
+    },
+    'fifo': {
+        'options': ['fifos', 'Pipes (fifo)', 'pipes', 'filesystem', 'monit.fifos', 'line'],
+        'lines': []
+    },
+    'program': {
+        'options': ['programs', 'Programs statuses', 'programs', 'applications', 'monit.programs', 'line'],
+        'lines': []
+    },
+    'process': {
+        'options': ['processes', 'Processes statuses', 'processes', 'applications', 'monit.services', 'line'],
+        'lines': []
+    },
+    'process_uptime': {
+        'options': ['processes uptime', 'Processes uptime', 'seconds', 'applications', 'monit.process_uptime', 'line', 'hidden'],
+        'lines': []
+    },
+    'process_threads': {
+        'options': ['processes threads', 'Processes threads', 'threads', 'applications', 'monit.process_threads', 'line'],
+        'lines': []
+    },
+    'process_children': {
+        'options': ['processes childrens', 'Child processes', 'childrens', 'applications', 'monit.process_childrens', 'line'],
+        'lines': []
+    },
+    'host': {
+        'options': ['hosts', 'Hosts', 'hosts', 'network', 'monit.hosts', 'line'],
+        'lines': []
+    },
+    'host_latency': {
+        'options': ['hosts latency', 'Hosts latency', 'milliseconds/s', 'network', 'monit.host_latency', 'line'],
+        'lines': []
+    },
+    'net': {
+        'options': ['interfaces', 'Network interfaces and addresses', 'interfaces', 'network', 'monit.networks', 'line'],
+        'lines': []
+    },
+}
+
+class Service(UrlService):
+    def __init__(self, configuration=None, name=None):
+        UrlService.__init__(self, configuration=configuration, name=name)
+        self.url = self.configuration.get('url', "http://127.0.0.1:2812/_status?format=xml&level=full")
+        self.order = ORDER
+        self.definitions = CHARTS
+
+    def parse(self, data):
+        try:
+            xml = ET.fromstring(data)
+        except ET.ParseError:
+            self.debug('{0} is not a vaild XML page. Please add "_status?format=xml&level=full" to monit URL.'.format(self.url))
```

Comment:
Building on the comment above, this should probably be made more generic.

Even if you don't go with hard-coding the path and query string, this should not suggest adding them, because there are plenty of other reasons this error condition could fail.

If you _really_ wanted to do this and not hard-code the URL, then you should be checking the URL itself, and deciding whether to use this or a more generic message based on that.

Reply:
Also, I'd say this should be more than just a debug message.  It means that you can't collect data at all, so an error or warning makes more sense (while the python.d.plugin code will print an error message if the check fails, having an explanation for _why_ it failed without having to go through manually debugging the plugin is much friendlier for users).

Reply:
Since it is `ET.ParseError` `is not a vaild XML page` seem enough, but yes, why not to log the error.

`except ET.ParseError as error:`

Reply:
>  It means that you can't collect data at all, so an error or warning makes more sense (while the python.d.plugin code will print an error message if the check fails, having an explanation for why it failed without having to go through manually debugging the plugin is much friendlier for users).

Yep, will do. But only one question. Is it ok to use self.warning() for this or there are any limitations for this? I looked into other modules and didn't find any usage of "warning" method. Or this should be an error (as it's "incorrect configuration")?

Please, clarify.

Reply:
i believe `/_status` thing in `apache`, `nignx`, `hapoxy` is configurable, then it shouldn't be static.

what about `monit`? does it always `_status?format=xml&level=full` ?

Reply:
Yes, for some reason nobody use `warning` :smile: Yes, it is safe. 
But this should be an error.

Reply:
Done it. Small example:

```
# head -n 10 log
2018-08-05 12:52:15: python.d INFO: plugin: main: Using python 3
2018-08-05 12:52:16: python.d INFO: monit: localhost: check() => [OK]
2018-08-05 12:52:16: python.d ERROR: monit: remote: Url: http://10.177.20.1:2812/_status?format=xml&level=full. Error: HTTPConnectionPool(host='10.177.20.1', port=2812): Max retries exceeded with url: /_status?format=xml&level=full (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fb7200be780>: Failed to establish a new connection: [Errno 111] Connection refused',))
2018-08-05 12:52:16: python.d INFO: monit: remote: check() => [FAILED]
> /usr/lib/x86_64-linux-gnu/netdata/python.d/monit.chart.py(92)parse()
-> self.error("URL {0} didn't return a vaild XML page. Please check your settings.".format(self.url))
(Pdb) 2018-08-05 12:52:17: python.d ERROR: monit: error: URL http://localhost:19999 didn't return a vaild XML page. Please check your settings.
2018-08-05 12:52:17: python.d INFO: monit: error: check() => [FAILED]
CHART netdata.runtime_monit_local '' 'Execution time for monit_local' 'ms' 'python.d' netdata.pythond_runtime line 145000 1
```

Plus also updated python.d/README.md and conf.d/monit.conf.